lib/fast_gettext/translation.rb

```diff
@@ -71,18 +60,23 @@ def Nn_(*keys)
       keys
     end
 
-    def ns_(*args, &block)
-      translation = n_(*args, &block)
+    def ns_(*keys)
+      translation = n_(*keys){ nil }
+      return translation.split(NAMESPACE_SEPARATOR).last if translation
 
-      # block is called once again to compare result TODO: this is bad
-      block && translation == block.call ? translation : translation.split(NAMESPACE_SEPARATOR).last
+      return yield if block_given?
+
+      FastGettext::PluralizationHelper.fallback(*keys).split(NAMESPACE_SEPARATOR).last
     end
     alias :nsgettext :ns_
 
-    def np_(context, key, *args)
-      options = (args.last.is_a? Hash) ? args.pop : {}
-      nargs = ["#{context}#{options[:separator]||CONTEXT_SEPARATOR}#{key}"] + args
-      n_(*nargs){nil} or (block_given? ? yield : key)
+    def np_(context, *keys)
```

Comment:
a keyword-arg `separator: nil` would be nice here to get rid of the options hash logic